File: src/classes/kftl/kftl_none/kftl-none-statement-line.ts
'use strict'

import type { KFTLRequestMap } from '../kftl-request-map'
import { KFTLStatementLine } from '../kftl-statement-line'
import { KFTLStatementLineConstructorFactory } from '../kftl-statement-line-constructor-factory'
import type { KFTLStatementLineContext } from '../kftl-statement-line-context'

export class KFTLNoneStatementLine extends KFTLStatementLine {
    constructor(line_text: string, context: KFTLStatementLineContext) {
        super(line_text, context)
        context.set_is_next_prototype(true)
        context.set_is_this_prototype(true)
        context.set_next_statement_line_constructor((line_text: string, context: KFTLStatementLineContext) => KFTLStatementLineConstructorFactory.get_instance().generate_none_constructor(line_text)(line_text, context))
        context.set_next_statement_line_target_id(context.get_this_statement_line_target_id())
    }

    async apply_this_line_to_request_map(_requet_map: KFTLRequestMap): Promise<void> {
        return
    }

    get_label_name(_context: KFTLStatementLineContext): string {
        return "**********"
    }
}

File: src/server/gkill/api/kftl/kftl_none_statement_line.go
package kftl

import (
	"context"
	"fmt"
)

// kftlNoneStatementLine represents a "done" state — no more processing for this entity.
// If a non-empty line arrives, it is an error.
// Mirrors: src/classes/kftl/kftl_none/kftl-none-statement-line.ts
type kftlNoneStatementLine struct {
	lineText string
	ctx      *KFTLStatementLineContext
}

func newKFTLNoneStatementLine(lineText string, ctx *KFTLStatementLineContext) *kftlNoneStatementLine {
	ctx.ThisIsPrototype = true
	ctx.NextIsPrototype = true
	targetID := ctx.ThisStatementLineTargetID
	ctx.NextStatementLineTargetID = &targetID
	// Next line is also None (but re-evaluated by factory for meta-info patterns)
	ctx.NextStatementLineConstructor = ctx.factory.generateNoneConstructor(ctx.NextStatementLineText)
	return &kftlNoneStatementLine{lineText: lineText, ctx: ctx}
}

func (l *kftlNoneStatementLine) ApplyThisLineToRequestMap(_ context.Context, _ *KFTLRequestMap) error {
	if l.lineText != "" {
		return fmt.Errorf("unexpected non-empty line in none-state: %q", l.lineText)
	}
	return nil
}

func (l *kftlNoneStatementLine) GetLabelName() string                 { return "none" }
func (l *kftlNoneStatementLine) GetContext() *KFTLStatementLineContext { return l.ctx }
func (l *kftlNoneStatementLine) GetStatementLineText() string         { return l.lineText }
